lib/error/try.h

```diff
@@ -0,0 +1,59 @@
+#ifndef LIB_ERROR_TRY_H_
+#define LIB_ERROR_TRY_H_
+
+#include "lib/base/paste.h"
+
+#define _TRY(expr, tmp)      \
+  {                          \
+    const auto tmp = (expr); \
```

Comment:
should this be const auto& tmp? I guess Status and StatusOr are always copiable, and we're ok with that copy? or assume that most of the time it's the value returned by a function, so return value optimization?

Reply:
const auto& tmp would depend on reference lifetime extension, correct? Not that this is bad, I just tend to get nervous when taking references to temporaries!

StatusOr actually is not guaranteed to be copiable if it contains a move only type. However TRY() should only interact with Status and not StatusOr. I was counting on RVO for the assignment to 'tmp', however I don't know how to avoid a copy in the later return statement. Will a reference make a difference? Copying a non-OK status is no-trivial so we should try to avoid it.